controllers/nstemplateset/client_test.go

```diff
@@ -0,0 +1,199 @@
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(role, devNs), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, changed)
+		assertObject(t, fakeClient, false)
+	})
+
+	t.Run("when only DBaaSTenant is supposed to be applied but the group for DBaaS is not present", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+		_, err := client.NewApplyClient(fakeClient, scheme.Scheme).Apply(objects(role, devNs), additionalLabel)
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(dBaaSTenant), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.False(t, changed)
+		assertObject(t, fakeClient, false)
+	})
+
+	t.Run("when only DBaaSTenant is supposed to be applied and the group for DBaaS is present", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+		apiClient.AvailableAPIGroups = append(apiClient.AvailableAPIGroups, metav1.APIGroup{Name: "dbaas.redhat.com"})
+		_, err := client.NewApplyClient(fakeClient, scheme.Scheme).Apply(objects(role, devNs), additionalLabel)
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(dBaaSTenant), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, changed)
+		assertObject(t, fakeClient, true)
+	})
+}
+
+func objects(objects ...runtimeclient.Object) []runtimeclient.Object {
+	var objs []runtimeclient.Object
+	for i := range objects {
+		objs = append(objs, objects[i].DeepCopyObject().(runtimeclient.Object))
+	}
+	return objs
+}
+
+func assertObject(t *testing.T, client *test.FakeClient, expectDBaaSTenant bool) {
+	AssertThatRole(t, "john-dev", "edit-john", client).
+		Exists(). // created
+		HasLabel(toolchainv1alpha1.ProviderLabelKey, toolchainv1alpha1.ProviderLabelValue).
+		HasLabel(toolchainv1alpha1.OwnerLabelKey, "john").
+		HasLabel("foo", "bar")
+	AssertThatNamespace(t, "john-dev", client).
+		HasLabel(toolchainv1alpha1.ProviderLabelKey, toolchainv1alpha1.ProviderLabelValue).
+		HasLabel(toolchainv1alpha1.OwnerLabelKey, "john").
+		HasLabel(toolchainv1alpha1.TypeLabelKey, "dev").
+		HasLabel("foo", "bar")
+	dBaaSTenant := &dbaasv1alpha1.DBaaSTenant{}
+	if expectDBaaSTenant {
+		AssertObject(t, client, "", "john-dev-tenant", dBaaSTenant, func() {
+			assert.Contains(t, dBaaSTenant.Labels, toolchainv1alpha1.ProviderLabelKey)
+			assert.Equal(t, toolchainv1alpha1.ProviderLabelValue, dBaaSTenant.Labels[toolchainv1alpha1.ProviderLabelKey])
+			assert.Contains(t, dBaaSTenant.Labels, toolchainv1alpha1.OwnerLabelKey)
+			assert.Equal(t, "john", dBaaSTenant.Labels[toolchainv1alpha1.OwnerLabelKey])
+			assert.Contains(t, dBaaSTenant.Labels, "foo")
+			assert.Equal(t, "bar", dBaaSTenant.Labels["foo"])
+		})
+	} else {
+		// should not exist
+		AssertObject(t, client, "john-dev", "john-dev-tenant", dBaaSTenant, nil)
```

Comment:
we should have an `AssertObjectDoesNotExist()` func instead of passing `nil` as the last argument here. But that can be addressed in a separate PR.

Reply:
see https://github.com/codeready-toolchain/member-operator/pull/347 (with `AssertObjectNotFound`, which is a shorter name and matches with the `NotFound` error)

Reply:
will be addressed in your PR - as part of solving the conflicts